Targeted documentation updates in autograd.functional

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #72111

For vectorize flag:
- Advertises the use of functorch

For autograd.functional.jvp:
- Advertises the use of functorch and the low-level jvp API, both of
which will be more performant than the double backprop trick.

Test Plan:
- view docs

Differential Revision: [D33918065](https://our.internmc.facebook.com/intern/diff/D33918065)

zou3519 committed:
Targeted documentation updates in autograd.functional

For vectorize flag:
- Advertises the use of functorch

For autograd.functional.jvp:
- Advertises the use of functorch and the low-level jvp API, both of
which will be more performant than the double backprop trick.

Test Plan:
- view docs

[ghstack-poisoned]

zou3519 committed:
Update on "Targeted documentation updates in autograd.functional"


For vectorize flag:
- Advertises the use of functorch

For autograd.functional.jvp:
- Advertises the use of functorch and the low-level jvp API, both of
which will be more performant than the double backprop trick.

Test Plan:
- view docs

[ghstack-poisoned]

albanD reviewed (approved):
SGTM

zou3519 commented:
@zou3519 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33918065).

zou3519 committed:
Update on "Targeted documentation updates in autograd.functional"


For vectorize flag:
- Advertises the use of functorch

For autograd.functional.jvp:
- Advertises the use of functorch and the low-level jvp API, both of
which will be more performant than the double backprop trick.

Test Plan:
- view docs

Differential Revision: [D33918065](https://our.internmc.facebook.com/intern/diff/D33918065)

[ghstack-poisoned]

zou3519 commented:
@zou3519 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33918065).

albanD commented on torch/autograd/functional.py:
```diff
@@ -323,6 +323,13 @@ def jvp(func, inputs, v=None, create_graph=False, strict=False):
+        the backward (sometimes called the double backwards trick). This is not
+        the most performant way of computing the jvp. Please consider using
+        `functorch's jvp <https://github.com/pytorch/functorch#jvp>`_
+        or the low-level forward-mode AD API instead.
```
You can link to https://pytorch.org/docs/master/autograd.html#forward-mode-automatic-differentiation here

zou3519 replied:
Yup, it took me a while to figure out how to actually get that cross-reference to work.

zou3519 committed:
Update on "Targeted documentation updates in autograd.functional"


For vectorize flag:
- Advertises the use of functorch

For autograd.functional.jvp:
- Advertises the use of functorch and the low-level jvp API, both of
which will be more performant than the double backprop trick.

Test Plan:
- view docs

Differential Revision: [D33918065](https://our.internmc.facebook.com/intern/diff/D33918065)

[ghstack-poisoned]

zou3519 commented:
@zou3519 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33918065).

zou3519 commented:
I ended up building the documentation locally because I think there is something wrong with the docs build job in CI. It looks fine:
![image](https://user-images.githubusercontent.com/5652049/152078331-14574a65-7ab8-4d63-be0a-e572d477e6d5.png)
